packages/core/rendering/core-rendering-server-internal/src/rendering_service.tsx

```diff
@@ -101,14 +103,36 @@ export class RenderingService {
     const buildNum = env.packageInfo.buildNum;
     const basePath = http.basePath.get(request);
     const { serverBasePath, publicBaseUrl } = http.basePath;
+
+    let settingsUserValues: Record<string, UserProvidedValues> = {};
```

Comment:
NIt: Can we make this one array `userValues = [{}, {}, {}]`?  then the whole reading from promise can be simplified to `userValues = await Promise.all(...)`.